File: Original-Artifact/app/src/main/java/com/example/freeman_option1/AddItem.java
package com.example.freeman_option1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class AddItem extends AppCompatActivity {

    InventoryDatabase inventoryDatabase;

    Button addItem;

    EditText editTextItemCount, editTextItemName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_item);

        addItem = (Button) findViewById(R.id.addItemButton);

        editTextItemCount = (EditText) findViewById(R.id.editCount);
        editTextItemName = (EditText) findViewById(R.id.editName);

        addItem.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View V){
                String itemName = editTextItemName.getText().toString();
                String itemCount = editTextItemCount.getText().toString();

                inventoryDatabase = new InventoryDatabase(getBaseContext());
                try{
                    int x = Integer.parseInt(itemCount);
                }catch(NumberFormatException e){
                    Toast.makeText(getBaseContext(), "Item Count must be an integer", Toast.LENGTH_LONG).show();
                    return;
                }
                boolean success = inventoryDatabase.insertItem(itemName, itemCount);
                if (success){
                    Toast.makeText(getBaseContext(), "Item Inserted successfully", Toast.LENGTH_LONG).show();
                    editTextItemName.setText("");
                    editTextItemCount.setText("");
                    Intent intent = new Intent (getBaseContext(), GridViewLayout.class);
                    startActivity(intent);
                }else{
                    Toast.makeText(getBaseContext(), "Item insertion failure", Toast.LENGTH_LONG).show();
                };
            }
        });


    }

    public void goToGridViewLayout (View view){
        Intent intent = new Intent (this, GridViewLayout.class);
        startActivity(intent);
    }
}

File: Data Structures and Algorithms/app_inventory_management/src/main/java/com/example/freeman_option1/AddItem.kt
package com.example.freeman_option1

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

/**
 * Add item
 *
 * @constructor Create empty Add item
 */
class AddItem : AppCompatActivity() {
    var inventoryDatabase: InventoryDatabase? = null
    var cache: InventoryCache<String, Item>? = null

    var addItem: Button? = null

    var editTextItemCount: EditText? = null
    var editTextItemName: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_item)

        addItem = findViewById<View>(R.id.addItemButton) as Button

        editTextItemCount = findViewById<View>(R.id.editCount) as EditText
        editTextItemName = findViewById<View>(R.id.editName) as EditText

        addItem!!.setOnClickListener(View.OnClickListener {
            val itemName = editTextItemName!!.text.toString()
            val itemCount = editTextItemCount!!.text.toString()

            inventoryDatabase = InventoryDatabase(baseContext)
            try {
                itemCount.toInt()
            } catch (e: NumberFormatException) {
                Toast.makeText(baseContext, "Item Count must be an integer", Toast.LENGTH_LONG).show()
                return@OnClickListener
            }

            val success = inventoryDatabase!!.insertItem(itemName, itemCount)
            val item = Item(itemName, itemCount);
            cache!!.put(itemName, item);

            if (success) {
                Toast.makeText(baseContext, "Item Inserted successfully", Toast.LENGTH_LONG).show()
                editTextItemName!!.setText("")
                editTextItemCount!!.setText("")
                val intent = Intent(baseContext, GridViewLayout::class.java)
                startActivity(intent)
            } else {
                Toast.makeText(baseContext, "Item insertion failure", Toast.LENGTH_LONG).show()
            }
        })
    }

    /**
     * Go to grid view layout
     *
     */
    fun goToGridViewLayout() {
        val intent = Intent(this, GridViewLayout::class.java)
        startActivity(intent)
    }
}